### ISSUE TITLE
Performance concerns

### ISSUE THREAD

**Issue by xxalpha** [monster860/FastDMM#19]
I think this needs some optimization so it is not as CPU intensive as it currently is.

With Dreammaker, you can only max CPU usage when moving the camera around at 50% zoom. With fastDMM, the CPU usage is at a constant maximum when you're just a little zoomed out, even if you are not moving the camera.

When you zoom in all the way and don't move the camera the CPU usage is lower, but it is not zero as is the case with DM.

When only space is on the screen, the CPU usage is lower, but at a certain zoom threshold it also maxes out.

Note that I am not running this on a toaster PC.

**Comment by monster860** [comment]
The rendering code is horribly optimized.

**Comment by Chnkr** [comment]
The regexes themselfs are also terribly optimized.

**Comment by PJB3005** [comment]
Yep there's still tons of room for improvements around the place.
